lib/src/core/advice.ts

```diff
@@ -9,11 +9,29 @@ export abstract class Advice {
     if (target.__woven__) {
       return (metadata.method.result = target.bind(this.context, metadata).apply(null, metadata.method.args));
     }
-      
+
     if (metadata.method.proceed) {
       return (metadata.method.result = target.apply(metadata.method.context, metadata.method.args));
-    } 
-        
+    }
+
+    return metadata.method.result;
+  }
+}
+
+export abstract class AsyncAdvice {
+  constructor(public context: Object, public advice: Function) {}
+
+  public async abstract wove(target: Function, metadata: Metadata);
+
+  public async invoke(target: any, metadata: Metadata) {
+    if (target.__woven__) {
+      return await (metadata.method.result = target.bind(this.context, metadata).apply(null, metadata.method.args));
```

Comment:
I'm going over the code again. I'm wondering, why don't we just do this here

```
if (target.__woven__) {
    metadata.method.result = await target.bind(this.context, metadata).apply(null, metadata.method.args);
} else if (metadata.method.proceed) {
    metadata.method.result = await target.apply(metadata.method.context, metadata.method.args));
}

return metadata.method.result;
```

Might be a a good reason for that though. But if not, I think this is more readable than what we have now. What do you think?